sites/fast-website/src/app/components/side-navigation/side-navigation.ts

```diff
@@ -14,21 +19,30 @@ export class SideNavigation extends FASTElement {
     @observable
     public sectionArray: HTMLElement[] = [];
 
+    @observable
+    public objArray: sectionData[];
+
     public socialData: CommunityContentPlacementData[] = communityContentPlacementData.filter(
         x => x.header !== "Github"
     );
 
     loadScrollers() {
         this.sectionArray = Array.from(document.querySelectorAll("section[id]"));
 
+        this.objArray = this.sectionArray.map(x => {
+            return { section: x.id, value: 0 };
+        });
+
         let observer = new IntersectionObserver(
             entries => {
                 entries.forEach(entry => {
-                    const areaOnScreen =
-                        entry.intersectionRatio * entry.boundingClientRect.height;
-                    if (areaOnScreen > 0.5 * window.innerHeight) {
-                        this.currentSection = entry.target.id;
-                    }
+                    const intersectingSection: sectionData = this.objArray.find(
+                        ({ section }) => section === entry.target.id
+                    ) as sectionData;
+                    intersectingSection.value = entry.intersectionRatio;
+                    this.currentSection = this.objArray.reduce((prev, current) =>
+                        prev.value > current.value ? prev : current
```

Comment:
Does this do what we want? I think this will set the "current section" to be the one at the highest index that is the most visible.  ie. if all sections are visible the last one is the current one.

Reply:
I have added a function that checks if the page is scrolling up or down. If the page is scrolling down it runs through the array in the standard order meaning that if all sections are visible the one lowest on the page is the current one. If the page is scrolling up the array is reversed meaning that if all sections are visible the one highest on the page one is the current one.